samples/kotlin-android-app-destinations/src/main/java/com/segment/analytics/destinations/plugins/FirebaseDestination.kt

```diff
@@ -0,0 +1,236 @@
+                "total" to Param.VALUE,
+                "revenue" to Param.VALUE,
+                "order_id" to Param.TRANSACTION_ID,
+                "currency" to Param.CURRENCY,
+                "products" to Param.ITEM_LIST
+        )
+
+        private val PRODUCT_MAPPER: Map<String, String> = mapOf(
+                "category" to Param.ITEM_CATEGORY,
+                "product_id" to Param.ITEM_ID,
+                "id" to Param.ITEM_ID,
+                "name" to Param.ITEM_NAME,
+                "price" to Param.PRICE,
+                "quantity" to Param.QUANTITY
+        )
+    }
+
+    override fun setup(analytics: Analytics) {
+        super.setup(analytics)
+
+        firebaseAnalytics = FirebaseAnalytics.getInstance(context)
+    }
+
+    override fun identify(payload: IdentifyEvent): BaseEvent? {
+        var returnPayload = super.identify(payload)
+
+        firebaseAnalytics.setUserId(payload.userId)
+
+        payload.traits.let {
+            for ((traitKey, traitValue) in it) {
+                val updatedTrait = makeKey(traitValue.toString())
+                firebaseAnalytics.setUserProperty(traitKey, updatedTrait)
+            }
+        }
+
+        return returnPayload
+    }
+
+    override fun track(payload: TrackEvent): BaseEvent? {
+        var returnPayload = super.track(payload)
+
+        // Clean the eventName up
+        var eventName = payload.event
+        if (EVENT_MAPPER.containsKey(eventName)) {
+            eventName = EVENT_MAPPER[eventName].toString()
+        } else {
+            eventName = makeKey(eventName)
+        }
+
+        payload.properties.let {
+            val formattedProperties = formatProperties(it)
+            firebaseAnalytics.logEvent(eventName, formattedProperties)
+            analytics.log("firebaseAnalytics.logEvent($eventName, $formattedProperties")
+        }
+
+        return returnPayload
+    }
+
+    override fun screen(payload: ScreenEvent): BaseEvent? {
+        var returnPayload = super.screen(payload)
+
+        val tempActivity = activity
+        if (tempActivity != null) {
+            firebaseAnalytics.setCurrentScreen(tempActivity, payload.name, null);
+        }
+
+        return returnPayload
+    }
+
+
+    // AndroidActivity Methods
+    override fun onActivityResumed(activity: Activity?) {
+        super.onActivityResumed(activity)
+
+        try {
+            val packageManager = activity?.packageManager ?: return
+
+            packageManager.getActivityInfo(activity.componentName, PackageManager.GET_META_DATA).let {
+                it.loadLabel(packageManager).toString().let {
+                    firebaseAnalytics.setCurrentScreen(activity, it, null)
+                    analytics.log("firebaseAnalytics.setCurrentScreen(activity, $it, null")
+                }
+            }
+        } catch (exception: PackageManager.NameNotFoundException) {
+            throw AssertionError("Activity Not Found: " + exception.toString())
```

Comment:
not sure if we want to be throwing an error or simply logging it

Reply:
We were throwing on the previous destination, do we want to change behavior? Down to log it.